### PR TITLE
USAGOV-1995 new scam tool w3 c markup validation errors

### PR DESCRIPTION
**Pull request by cwacht**
## Jira Task

https://cm-jira.usa.gov/browse/USAGOV-1995

## Description

This change resolves HTML validation errors caught by running this:
https://validator.w3.org/nu/?doc=https%3A%2F%2Fwww.usa.gov%2Fwhere-report-scams%2Fwhere-did-scam-take-place
This change also cleans up the sidenav code and makes it more reusable so if we decide to create custom sidenavs in the future, it will be simple to use the existing template rather than creating new templates that try to generate the same sidenav structure on their own.

## Testing Instructions

- Visit localhost/where-report-scams/where-did-scam-take-place
- View Page Source (Dont use dev tools to inspect since it automatically corrects some invalid HTML)
- Copy all the HTML
- Visit https://validator.w3.org/#validate_by_input
- Paste the copied HTML into the textarea
- Click the Check button
- Confirm that there are no errors or warnings beyond the following:
  - Bad values for `ID` attributes on Labels and `for` attributes on Inputs.
    - These have been fixed in content on Prod
  - Duplicate attribute class.
    - This is being resolved by another ticket

## Type of Changes

- [ ] New Feature
- [ ] Bugfix
- [x] Frontend (Twig, Sass, JS)
  - Add screenshot showing what it should look like
- [ ] Drupal Config (requires "drush cim")
- [ ] New Modules (requires rebuild)
- [ ] Documentation
- [ ] Infrastructure
  - [ ] CMS
  - [ ] WAF
  - [ ] WWW
  - [ ] Egress
  - [ ] Tools
  - [ ] Cron
- [ ] Other

### Change Requirements

- [ ] Requires New Documentation (Link: {})
- [ ] Requires New Config
- [ ] Requires New Content

### Validation Steps

- Test instruction 1
- Test instruction 2
- Test instruction 3

## Security Review

- [ ] Adds/updates software (including a library or Drupal module)
- [ ] Communication with external service
- [ ] Changes permissions or workflow
- [ ] Requires SSPP updates

## Reviewer Reminders

- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions

Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
2. Find the commit of this pull request.
3. Build and deploy the changes.
4. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.

### CODE REVIEW

**Comment by omerida** [comment]
Confirming the only validation errors when not logged in are the ones indicated above.

File: web/modules/custom/usagov_menus/src/Plugin/Block/SidebarFirstBlock.php [hunk 1]
```diff
@@ -108,67 +111,22 @@ private function buildFromMenu(string $menuID): array {
...
-    $items = $this->getMenuTreeItems($menuID, $crumbs, $active, closeLastTrail: TRUE);
-
-    $node = $this->routeMatch->getParameter('node');
-
-    return $this->renderItems($items, $active, leaf: [
-      'url' => $this->request->getPathInfo(),
-      'title' => $node->getTitle(),
-    ]);
-  }
-
```
Comment (omerida) [inline web/modules/custom/usagov_menus/src/Plugin/Block/SidebarFirstBlock.php:171]: Removing this changes how the sidebar on state pages displays. It's why I had the `closeLastTrail` flag. On prod, the sidebar only shows the current state as the only sub item.

image

This branch shows all the subchildren without the State being viewed as the active link. 

image

I assume we at least want to insert the current active link into that. Displaying the other siblings makes the nav more consistent with how it works across the site but do we need to run that change by UX or content folks?

**Review by omerida — CHANGES_REQUESTED** [review]
Sidebar on English states changed, was that intentional?

File: web/modules/custom/usagov_menus/src/Plugin/Block/SidebarFirstBlock.php (hunk 1 above)
Comment (cwacht) [reply web/modules/custom/usagov_menus/src/Plugin/Block/SidebarFirstBlock.php:171]: Thanks for catching this. My local environment was showing the correct sidenav for these pages, but it looks like I needed to shutdown and restart my environment to see it properly.
I've incorporated $closeLastTrail back into the code that is executed for state pages and that makes it match prod again.

**Review by omerida — APPROVED** [review]
I confirmed the validation errors are gone. State sidebar is correct and this also removes building a sidebar for wizard pages that was hidden anyway, which is a nice bonus.

**Pull request merged**